src/pybamm/experiment/step/base_step.py

```diff
@@ -266,6 +266,84 @@ def default_duration(self, value):
         else:
             return 24 * 3600  # one day in seconds
 
+    @staticmethod
+    def default_period():
+        return 60.0  # seconds
+
+    def default_time_vector(self, tf, t0=0):
+        if self.period is None:
+            period = self.default_period()
+        else:
+            period = self.period
+        npts = max(int(round(np.abs(tf - t0) / period)) + 1, 2)
+
+        return np.linspace(t0, tf, npts)
+
+    def setup_timestepping(self, solver, tf, t_interp=None):
+        """
+        Setup timestepping for the model.
+
+        Parameters
+        ----------
+        solver: :class`pybamm.BaseSolver`
+            The solver
+        tf: float
+            The final time
+        t_interp: np.array | None
+            The time points at which to interpolate the solution
+        """
+        if solver._supports_interp:
```

Comment:
We don't do this well in general, but private methods should only be accessed from within the same file/class - in this case each `Solver` class should implement a property `supports_interp` in order to make it truly read-only.